src/rules/jsxCurlySpacingRule.ts

```diff
@@ -0,0 +1,142 @@
+
+const OPTION_ALWAYS = "always";
+const OPTION_NEVER = "never";
+const SPACING_VALUES = [OPTION_ALWAYS, OPTION_NEVER];
+/* tslint:disable:object-literal-sort-keys */
+const SPACING_OBJECT = {
+    type: "string",
+    enum: SPACING_VALUES,
+};
+/* tslint:enable:object-literal-sort-keys */
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "jsx-curly-spacing",
+        description: "Checks JSX curly braces spacing",
+        optionsDescription: Lint.Utils.dedent`
+            One of the following two options must be provided:
+
+            * \`"${OPTION_ALWAYS}"\` requires JSX attributes to have spaces between curly braces
+            * \`"${OPTION_NEVER}"\` requires JSX attributes to NOT have spaces between curly braces`,
+        options: {
+            type: "array",
+            items: [SPACING_OBJECT],
+            minLength: 1,
+            maxLength: 1,
+        },
+        optionExamples: [
+            `[true, "${OPTION_ALWAYS}"]`,
+            `[true, "${OPTION_NEVER}"]`,
+        ],
+        type: "style",
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_NO_ENDING_SPACE = (tokenStr: string) => {
+        return `A space is required before ${tokenStr}`;
+    };
+
+    public static FAILURE_NO_BEGINNING_SPACE = (tokenStr: string) => {
+        return `A space is required after ${tokenStr}`;
+    };
+
+    public static FAILURE_FORBIDDEN_SPACES_BEGINNING = (tokenStr: string) => {
+        return `There should be no space after ${tokenStr}`;
+    };
+
+    public static FAILURE_FORBIDDEN_SPACES_END = (tokenStr: string) => {
+        return `There should be no space before ${tokenStr}`;
+    };
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const walker = new JsxCurlySpacingWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(walker);
+    }
+}
+
+class JsxCurlySpacingWalker extends Lint.RuleWalker {
+    private always = this.hasOption(OPTION_ALWAYS) || false;
+
+    protected visitJsxExpression(node: ts.JsxExpression) {
+        this.validateBraceSpacing(node);
+
+        super.visitJsxExpression(node);
+    }
+
+    protected visitNode(node: ts.Node) {
+        if (node.kind === ts.SyntaxKind.JsxSpreadAttribute) {
+            this.validateBraceSpacing(node);
+        }
+
+        super.visitNode(node);
+    }
+
+    private validateBraceSpacing(node: ts.Node) {
+        const firstToken = node.getFirstToken();
+        const secondToken = node.getChildAt(1);
+        const lastToken = node.getLastToken();
+        const secondToLastToken = node.getChildAt(node.getChildCount() - 2);
+        const nodeStart = node.getStart();
+        const nodeWidth = node.getWidth();
+
+        function isExpressionMultiline(text: string) {
+            return /\n/.test(text.replace(/\/\*.*?\*\//g, ""));
```

Comment:
Just to double-check, why do you have to strip out comments?  Also might be best to store this regex as a module constant or as a class field, since it's used multiple times